pairinteraction/CMakeLists.txt

```
@@ -48,20 +48,39 @@ configure_file(version.h.in ${CMAKE_CURRENT_BINARY_DIR}/version.h)
 
 file(GLOB pairinteraction_SRCS *.cpp)
 list(REMOVE_ITEM pairinteraction_SRCS ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp)
+list(REMOVE_ITEM pairinteraction_SRCS ${CMAKE_CURRENT_SOURCE_DIR}/julia_hooks_pi.cpp)
 
 add_library(pireal SHARED ${pairinteraction_SRCS} ${wignerSymbols_SRCS})
 add_library(picomplex SHARED ${pairinteraction_SRCS} ${wignerSymbols_SRCS})
 set_target_properties(picomplex PROPERTIES COMPILE_DEFINITIONS "USE_COMPLEX")
 
+if(WITH_JULIA)
+  execute_process(COMMAND locate JlCxxConfig.cmake OUTPUT_VARIABLE JlCxxConfig_LOC)
```

`locate JlCxxConfig.cmake` won't work on Windows and on Linux it requires `mlocate` to be installed and the database to be up-to-date.  Instead something like
```suggestion
  find_file(JlCxxConfig_LOC NAMES JlCxxConfig.cmake)
```
should be used, but there you have to provide one or more paths where to search for `JlCxxConfig.cmake`, because it is not located in any of the standard paths.

---

It's possible to get the path using
```console
$ julia -e 'using CxxWrap; print(joinpath(dirname(CxxWrap.jlcxx_path), "cmake", "JlCxx"))'
```

---

We could set
```bash
export CMAKE_PREFIX_PATH="$(julia -e 'using CxxWrap; print(joinpath(dirname(CxxWrap.jlcxx_path), "cmake", "JlCxx"))')${CMAKE_PREFIX_PATH:+:${CMAKE_PREFIX_PATH}}"
```
although this doesn't really belong inside the Docker container.

---

There might be a simpler solution, see e.g. https://github.com/JuliaInterop/CxxWrap.jl/issues/115#issuecomment-437347226. We could try something like `get_target_property(JlCxx_location JlCxx::cxxwrap_julia LOCATION)`. However, I do not completely understand why we need this path, i.e. why we have to set `CMAKE_INSTALL_RPATH`?

---

You misunderstood.  Finding `JlCxxConfig.cmake` is required for `find_package(JlCxx)` to work.

But you are right, that the code following `find_package(JlCxx)` can be simplified with the stuff you linked.

---

I see. Could we get the Julia executable by calling `find_program` in CMake and use `execute_process` to call the executable and run your Julia code to obtain the path?